packages/server/src/fhir/operations/expand.ts

```diff
@@ -207,24 +208,50 @@ function processExpansion(systemExpressions: Expression[], expansionContains: Va
 
 const MAX_EXPANSION_SIZE = 1000;
 
-export function filterCodings(codings: Coding[], params: ValueSetExpandParameters): Coding[] {
+export function filterIncludedConcepts(
+  concepts: ValueSetComposeIncludeConcept[] | ValueSetExpansionContains[] | Coding[],
+  params: ValueSetExpandParameters,
+  system?: string
+): ValueSetExpansionContains[] {
   const filter = params.filter?.trim().toLowerCase();
+  const codings: Coding[] = flattenConcepts(concepts, { filter, system });
   if (!filter) {
     return codings;
   }
   return codings.filter((c) => c.display?.toLowerCase().includes(filter));
 }
 
-export async function expandValueSet(valueSet: ValueSet, params: ValueSetExpandParameters): Promise<ValueSet> {
-  let expandedSet: ValueSetExpansionContains[];
+function flattenConcepts(
+  concepts: ValueSetComposeIncludeConcept[] | ValueSetExpansionContains[] | Coding[],
+  options?: {
+    filter?: string;
+    system?: string;
+  }
+): Coding[] {
+  const result: Coding[] = [];
+  for (const concept of concepts) {
+    const system = (concept as Coding).system ?? options?.system;
+    if (!system) {
+      throw new Error('Missing system for Coding');
+    }
 
-  const expansion = valueSet.expansion;
-  if (expansion?.contains?.length && !expansion.parameter && expansion.total === expansion.contains.length) {
-    // Full expansion is already available, use that
-    expandedSet = filterCodings(expansion.contains, params);
-  } else {
-    expandedSet = await computeExpansion(valueSet, params);
```

Comment:
Moving this logic to the top of `computeExpansion()` is the primary change that fixes the error around including pre-expanded ValueSets